### PR TITLE
[ACTION] Add inReplyTo & references to gmail-send-email

### PR DESCRIPTION
**Pull request by jc-tzn**
The "References" and "In-Reply-To" headers are necessary to keep an email answer in the same thread as the original email. 

Related documentation:
- Gmail API: https://developers.google.com/gmail/api/guides/sending#sending_messages
- RFC 2822 standard: https://www.rfc-editor.org/rfc/rfc2822#appendix-A.2
- Nodemailer Mailcomposer: https://nodemailer.com/extras/mailcomposer/#e-mail-message-fields

### CODE REVIEW

**Comment by dylburger** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/gmail/actions/send-email/send-email.mjs [hunk 1]
```diff
@@ -63,6 +63,12 @@ export default {
       description: "Add any attachments you'd like to include as objects. The `key` should be the **filename** and the `value` should be the **url** for the attachment. The **filename** must contain the file extension (i.e. `.jpeg`, `.txt`) and the **url** is the download link for the file.",
       optional: true,
     },
+    inReplyTo: {
+      type: "string",
+      label: "In Reply To",
+      description: "Specify the message-id this email is replying to. To use this prop with async options please use `Gmail(Developer App)` `Send Email` component.",
```
Comment (andrewjschuang) [inline components/gmail/actions/send-email/send-email.mjs:69]:
```suggestion
      description: "Specify the `message-id` this email is replying to. To use this prop with `async options` please use `Gmail (Developer App)` `Send Email` component.",
```

Also we should remove the `replyTo` prop.

**Review by andrewjschuang — APPROVED** [review]
Looks great! I just added some nitpicks

**Review by andrewjschuang — CHANGES_REQUESTED** [review]
Actually, I copied the `message-id` from `async options` and pasted to the **Gmail** app and tried sending the email, but it said that the value is not a valid message ID.

The **Gmail Developer App** worked fine though.

**Comment by feyzullah** [comment]
> Actually, I copied the `message-id` from `async options` and pasted to the **Gmail** app and tried sending the email, but it said that the value is not a valid message ID.
> 
> The **Gmail Developer App** worked fine though.

Oops! This is again because of the insufficient scopes. Let me remove it from there and add directly to threadId.

**Comment by feyzullah** [comment]
Now this should be able to reply non-replied emails. If a replied message-id would be given Gmail API will return an error because `threadId` is not existing. We can remove this prop from Gmail app and keep it in Gmail (Developer App). What do you think @andrewjschuang?

**Comment by andrewjschuang** [comment]
Oh, so in the **Gmail** app `threadId` won't work? If so, yeah we can leave it only on the **Gmail Developer App**

**Comment by feyzullah** [comment]
> Oh, so in the **Gmail** app `threadId` won't work? If so, yeah we can leave it only on the **Gmail Developer App**

It works for the first message. If replying to a reply, it will fail. Then I am removing the prop from Gmail app

**Comment by andrewjschuang** [comment]
> It works for the first message. If replying to a reply, it will fail. Then I am removing the prop from Gmail app

Ok, I see. After testing a little bit more I have a few suggestions:

- `opts.inReplyTo` and `opts.references` are actually not needed (delete lines 96-97, 99-100).
- `inReplyTo` prop description in **Gmail** app: add that it must be the `message-id` of the first message sent in the thread.
- In the **Gmail** app, if the error is `Requested entity was not found` (`message-id` was not for the first message in the thread), change the error message to: **Unable to find email thread. `In Reply To` must be the `message-id` of the first message of the thread.**

Other than that looks great to me!

**Comment by feyzullah** [comment]
> 

@andrewjschuang I have applied second and third suggestions. However when I don't send `inReplyTo` and `refenreces`, sender side is able to cluster all under one thread ID but receiver side is not. The following happens for sender side;
sender

And for the receiver side;
receiver

Am I missing something here? Or is there some other config to be done?

**Comment by andrewjschuang** [comment]
> @andrewjschuang I have applied second and third suggestions. However when I don't send `inReplyTo` and `refenreces`, sender side is able to cluster all under one thread ID but receiver side is not.

Oh ok, sorry please keep it that way

**Review by andrewjschuang — APPROVED** [review]

**Pull request merged**